ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/resources/jts/orbspecific/XAResourceRecord.java

```diff
@@ -407,6 +401,12 @@ public void rollback() throws org.omg.CORBA.SystemException,
                                     destroyState = false;
                                 }
                                 break;
+                            case XAException.XAER_NOTA:
+                                if (_phaseTwoStarted) {
+                                    // rolled back previously and recovery completed
+                                    destroyState();
+                                    break;
+                                }
```

Comment:
Why do we need to fall through to the next arm of the case here?

Reply:
I notice that PR https://github.com/jbosstm/narayana/pull/2326 does the same but with the addition of a `throw new UNKNOWN();` for the XAER_RMFAIL case. Are both PRs candidates for merging?

Reply:
They are both candidates. This one is a bug, the other is for consideration of whether we want to go that way (on top of this). (I'll respond to the first comment in a follow-up)

Reply:
I am reverting the bug introduced in https://github.com/jbosstm/narayana/commit/7f41fcb996ad3492a8718d762b47ce5616499c82 whereby XAER_NOTA no longer throws an exception because it drops into the XAER_RMFAIL handling and breaks out of the switch and therefore bypasses https://github.com/jbosstm/narayana/blob/main/ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/resources/jts/orbspecific/XAResourceRecord.java#L413-L417

I added a comment to describe the potential issue hopefully then if someone adds another case they will be aware of this